test/test_queue.js

```diff
@@ -1297,6 +1297,124 @@ describe('Queue', function () {
       });
     });
 
+    it('should not retry a job that has been removed', function(done) {
+      queue = utils.buildQueue('retry a removed job');
+      queue.on('ready', function() {
+        var attempts = 0;
+        queue.process(function (job, jobDone) {
+          if(attempts === 0) {
+            attempts++;
+            jobDone(new Error('failed'));
+          } else {
+            jobDone();
```

Comment:
For simplicity here we could just throw and error or not returning anything, and skip the jobDone callback.